calculator/src/components/calculator.js

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import Buttons from "./buttons";
+import { evaluate } from "mathjs";
+
+class Calculator extends React.Component {
+  state = {
+    finalVal: "",
+    holder: "",
+    currentVal: "0",
+    prevVal: "",
+    opperater: "",
+    holdVal: "",
+    toggle: false,
+    clearSymbol: false
+  };
+
+  numButtonsHandler = e => {
+    let res = e.target.value;
+
+    if (this.state.opperater !== "") {
```

Comment:
Just a reminder that an empty string is a falsy value so you could shorten this line by simply saying  ` if(this.state.operator){ `   and that would mean the same thing as you've written here .